sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteGroupedRecordsToFiles.java

```diff
@@ -58,11 +57,11 @@ public void processElement(
       for (ElementT tableRow : element.getValue()) {
         if (writer.getByteSize() > maxFileSize) {
           writer.close();
-          writer = rowWriterFactory.createRowWriter(tempFilePrefix, element.getKey());
```

Comment:
this bug is captured after the modified writeDynamics test. the writer should be created after the `writer.getResult()` call. Otherwise causing `java.lang.IllegalStateException: Not yet closed`.